### ISSUE TITLE
Turn on strict rules for samples repo

### ISSUE THREAD

**Issue by yaelgoede** [Azure/azure-functions-nodejs-library#204]
When trying the eventhub trigger example from the official azure documentation you get typescript typing errors by default.

https://learn.microsoft.com/en-us/azure/azure-functions/functions-bindings-event-hubs-trigger?tabs=python-v2%2Cisolated-process%2Cnodejs-v4%2Cfunctionsv2%2Cextensionv5&pivots=programming-language-typescript#example

![image](https://github.com/Azure/azure-functions-nodejs-library/assets/44118611/76a3014c-1bc8-496b-b69a-380252c0d775)

![image](https://github.com/Azure/azure-functions-nodejs-library/assets/44118611/ca32633b-fb31-482f-9d55-32cc0b7628ee)

**Comment by ejizba** [comment]
Hi @yaelgoede that sample is pulled from [here](https://github.com/Azure/azure-functions-nodejs-samples/blob/main/ts/src/functions/eventHubTrigger2.ts) and we have a build that is passing at the moment. Could you share your tsconfig.json and let us know which version of `@azure/functions` you are on?

**Comment by yaelgoede** [comment]
@ejizba 
{
  "compilerOptions": {
    "module": "commonjs",
    "target": "es6",
    "outDir": "dist",
    "rootDir": ".",
    "sourceMap": true,
    "strict": true
  }
}

**Comment by ejizba** [comment]
Yeah if you turn off `strict` you should get a passing build. Some of those strict rules are more personal preference, although I can look into turning them on for our samples repository

**Comment by yaelgoede** [comment]
yeah fair enough. The Azure Functions Core Tools also don't use strict by default. Some consider it a best practise but I cannot find anything about this in the official typescript docs.

**Comment by ejizba** [comment]
Re-opening as I still want our team to discuss and potentially fix this
